distributed/protocol/skorch.py

```diff
@@ -0,0 +1,28 @@
+import cloudpickle
+import skorch
+from .serialize import dask_serialize, dask_deserialize
+
+@dask_serialize.register(skorch.NeuralNet)
+def serialize_skorch(x):
+    has_module = hasattr(x, "module_")
+    headers = {"has_module": has_module}
+    if has_module:
+        module = x.__dict__.pop("module_")
```

Comment:
Curious why `module_` can't be pickled on its own. Is there any more info on the issues encountered by leaving this?

Also any downside to (temporarily) modifying a user-provided object here?

Reply:
>Curious why module_ can't be pickled on its own. Is there any more info on the issues encountered by leaving this?

So module's is an interactively defined class on client so its namespace is often `__main__` . For eg. `__main__. MyModule`. 

Pickle has problems pickling when interactively defined classes  when they are set as an attributes of another object. as it tries to look up the class in the namespace. See [eg](https://github.com/dask/distributed/pull/5748#issuecomment-1028218301) for trace.  
By pickling it on its own we are able to serialize successfully

> Also any downside to (temporarily) modifying a user-provided object here?

The only side effect i can think of is if the class is redefined in the worker's name-space causing undefined behavior while de-serializing on the worker. I doubt that will really happen in real workflows. 

FWIW, I have added a test to verify that at-least for the class is the same after deserialization.

Reply:
Is this just an issue with `pickle`? Does `cloudpickle` run into this issue or does it work ok?